azurecontainerapps.ts

```diff
@@ -0,0 +1,519 @@
+
+    // Helper properties
+    private static telemetryHelper: TelemetryHelper;
+    //private static authHelper: AzureAuthenticationHelper;
+    private static appHelper: ContainerAppHelper;
+    private static registryHelper: ContainerRegistryHelper;
+
+    // Miscellaneous properties
+    private static imageToBuild: string;
+    private static runtimeStack: string;
+    private static ingress: string;
+    private static targetPort: string;
+    private static shouldUseUpdateCommand: boolean;
+
+    /**
+     * Initializes the helpers used by this task.
+     * @param disableTelemetry - Whether or not to disable telemetry for this task.
+     */
+    private static initializeHelpers(disableTelemetry: boolean) {
+        // Set up TelemetryHelper for managing telemetry calls
+        this.telemetryHelper = new TelemetryHelper(disableTelemetry);
+
+        // Set up AzureAuthenticationHelper for managing logging in and out of Azure CLI using provided service connection
+        // this.authHelper = new AzureAuthenticationHelper();
+
+        // Set up ContainerAppHelper for managing calls around the Container App
+        this.appHelper = new ContainerAppHelper(disableTelemetry);
+
+        // Set up ContainerRegistryHelper for managing calls around ACR
+        this.registryHelper = new ContainerRegistryHelper();
+    }
+
+    /**
+     * Validates the arguments provided to the task for supported scenarios.
+     * @throws Error if a valid combination of the support scenario arguments is not provided.
+     */
+    private static validateSupportedScenarioArguments() {
+
+        // Get the path to the application source to build and run, if provided
+        this.appSourcePath = core.getInput('appSourcePath', { required: false });
+
+        // Get the name of the ACR instance to push images to, if provided
+        this.acrName = core.getInput('acrName', { required: false });
+
+        // Get the previously built image to deploy, if provided
+        this.imageToDeploy = core.getInput('imageToDeploy', { required: false });
+
+        // Get the YAML configuration file, if provided
+        this.yamlConfigPath = core.getInput('yamlConfigPath', { required: false });
+
+        // Ensure that acrName is also provided if appSourcePath is provided
+        if (!util.isNullOrEmpty(this.appSourcePath) && util.isNullOrEmpty(this.acrName)) {
+            core.error(`The 'acrName' argument must be provided when the 'appSourcePath' argument is provided.`);
+            throw Error(`The 'acrName' argument must be provided when the 'appSourcePath' argument is provided.`);
+        }
+
+        // Ensure that one of appSourcePath, imageToDeploy, or yamlConfigPath is provided
+        if (util.isNullOrEmpty(this.appSourcePath) && util.isNullOrEmpty(this.imageToDeploy) && util.isNullOrEmpty(this.yamlConfigPath)) {
+            core.error(`One of the following arguments must be provided: 'appSourcePath', 'imageToDeploy', or 'yamlConfigPath'.`);
```

Comment:
Looks like we're missing a `throw Error` below this line, similar to what we're doing in the above if-statement block

Reply:
Updated.